FirebaseFunctions/Tests/Integration/IntegrationTests.swift

```diff
@@ -868,6 +868,225 @@ class IntegrationTests: XCTestCase {
+    ])
+    let result: [WeatherForecast] = try await stream.reduce([]) { $0 + [$1] }
+    XCTAssertEqual(
+      result,
+      [
+        WeatherForecast(location: Location(name: "Toronto"), temperature: 25, conditions: .snowy),
+        WeatherForecast(location: Location(name: "London"), temperature: 50, conditions: .rainy),
+        WeatherForecast(location: Location(name: "Dubai"), temperature: 75, conditions: .sunny),
+      ]
+    )
+  }
+
+  func testGenerateStreamContent_ComplexStreamResponse() async throws {
+    // TODO: Maybe a result type that is more complicated than `String`, like a `WeatherForecastReport`?
+    let callable: Callable<[Location], StreamResponse<WeatherForecast, String>> = functions
+      .httpsCallable("genStreamWeather")
+    let stream = callable.stream([
+      Location(name: "Toronto"),
+      Location(name: "London"),
+      Location(name: "Dubai"),
+    ])
+    var streamContents: [WeatherForecast] = []
+    var streamResult = ""
+    for try await response in stream {
+      switch response {
+      case let .message(message):
+        streamContents.append(message)
+      case let .result(result):
+        streamResult = result
+      }
+    }
+    XCTAssertEqual(
+      streamContents,
+      [
+        WeatherForecast(location: Location(name: "Toronto"), temperature: 25, conditions: .snowy),
+        WeatherForecast(location: Location(name: "London"), temperature: 50, conditions: .rainy),
+        WeatherForecast(location: Location(name: "Dubai"), temperature: 75, conditions: .sunny),
+      ]
+    )
+    XCTAssertEqual(streamResult, "Number of forecasts generated: 3")
+  }
+
+  func testGenerateStreamContent_ComplexStreamResponse_Functional() async throws {
+    // TODO: Maybe a result type that is more complicated than `String`, like a `WeatherForecastReport`?
+    let callable: Callable<[Location], StreamResponse<WeatherForecast, String>> = functions
+      .httpsCallable("genStreamWeather")
+    let stream = callable.stream([
+      Location(name: "Toronto"),
+      Location(name: "London"),
+      Location(name: "Dubai"),
+    ])
+    let result: (accumulatedMessages: [WeatherForecast], result: String) =
+      try await stream.reduce(([], "")) { partialResult, streamResponse in
+        switch streamResponse {
+        case let .message(message):
+          (partialResult.accumulatedMessages + [message], partialResult.result)
+        case let .result(result):
+          (partialResult.accumulatedMessages, result)
+        }
+      }
+    XCTAssertEqual(
+      result.accumulatedMessages,
+      [
+        WeatherForecast(location: Location(name: "Toronto"), temperature: 25, conditions: .snowy),
+        WeatherForecast(location: Location(name: "London"), temperature: 50, conditions: .rainy),
+        WeatherForecast(location: Location(name: "Dubai"), temperature: 75, conditions: .sunny),
+      ]
+    )
+    XCTAssertEqual(result.result, "Number of forecasts generated: 3")
+  }
```

Comment:
These two unit tests show how to process a stream's contents with the new `StreamResponse` enum in both a non-functional (for loop) and functional (reduce) way.

cc: @peterfriese, @paulb777, @andrewheard